File: pset1/mario/more/mario.c
//Implement a program that prints out a double half-pyramid of a specified height, per the below.


#include <stdio.h>
#include <cs50.h>

//Prototypes
void printTag(int i);
void printSpace(int size);

int main(void)
{
    int height;    //Container for User selection
    do
    {
        height = get_int("Height: ");
    }
    while (height < 0 || height > 23);

    for (int i = 0; i < height; i++)
    {
        printSpace(height - (i + 1));
        printTag(i);

        printSpace(2);
        printTag(i);
        printf("\n");

    }

}

//Prints each Symbol
void printTag(int i)
{
    for (int j = 0; j <= i; j++)
    {
        printf("#");
    }
}


//Prints each space
void printSpace(int size)
{
    for (int s = 0; s < size; s++)
    {
        printf(" ");
    }
}

File: pset6/mario/more/mario.py
# Implement a program that prints out a double half-pyramid of a specified height, per the below.

#Implement do while loop python style
height = -1
while height < 0 or height > 23:
    try:
        height = int(input("Height: "))
    except ValueError:
        height = -1
        pass

for i in range(1, height + 1):
    output = ""
    #create first half of pyramid
    for x in range(height - i):
        output += " "
    for x in range(i):
        output += "#"
    output += "  "
    #create second half of pyramid
    for x in range(i):
        output += "#"
    print(output)
